api/lib/oauth/oauth.js

```diff
@@ -16,7 +16,7 @@ module.exports = function(configFilePath) {
     var appId = reqPath.substring(startIndex, endIndex);
     if (!appId) {
       logger.error("Failed to get appId");
-      callback({ "message": "Failed to get appId" }, null);
+      callback({ "statusCode": HttpStatus.UNAUTHORIZED, "message": "Failed to get appId" }, null);
```

Comment:
If "appId" is not passed through in the request,  i.e. /v1/apps/policy/,  it should be 404 not found ..